includes/admin/class-senseilms-plugin-updater.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * File containing SenseiLMS_Plugin_Updater class.
+ *
+ * @package Sensei\Admin
+ * @since   $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Manages plugin updates by hooking into WordPress's plugins update system and querying SenseiLMS Store's API.
+ *
+ * It's a fallback to SenseiLMS_Licensing\SenseiLMS_Plugin_Updater from Sensei Pro. So it works when
+ * Sensei Pro is in a legacy version without this class.
+ *
+ * @since $$next-version$$
+ */
+class SenseiLMS_Plugin_Updater {
+
+	const CACHE_KEY_PREFIX = 'senseilms_plugin_updater_info__';
+	const CACHE_TTL        = 3600;
+
+	/**
+	 * Full qualified name for the plugin. Relative path from plugins directory. Example: 'akismet/akismet.php'.
+	 *
+	 * @var string
+	 */
+	private $plugin_full_name;
+
+	/**
+	 * Plugin slug name. Example: 'akismet'.
+	 *
+	 * @var string
+	 */
+	private $plugin_slug;
+
+	/**
+	 * Current version for the plugin. Example: '1.0.0'.
+	 *
+	 * @var string
+	 */
+	private $version;
+
+	/**
+	 * Private constructor.
+	 *
+	 * @param string $main_plugin_file_absolute_path Plugin's main file path.
+	 * @param string $version                        Plugin version.
+	 */
+	private function __construct( $main_plugin_file_absolute_path, $version ) {
+		$this->plugin_full_name = plugin_basename( $main_plugin_file_absolute_path );
+		$this->plugin_slug      = basename( $main_plugin_file_absolute_path, '.php' );
+		$this->version          = $version;
+	}
+
+	/**
+	 * Initialize the plugin updater.
+	 */
+	public static function init() {
+		add_action( 'plugins_loaded', [ __CLASS__, 'plugins_loaded' ] );
+	}
+
+	/**
+	 * Add hooks after the `plugins_loaded`, so we make sure Sensei Pro was
+	 * already loaded.
+	 */
+	public static function plugins_loaded() {
+		// Early return if license and updates are managed by WooCommerce.
+		if ( defined( 'SENSEI_COMPAT_PLUGIN' ) && SENSEI_COMPAT_PLUGIN ) {
+			return;
+		}
+
+		if ( class_exists( '\Sensei_Pro_Interactive_Blocks\Setup_Context' ) ) {
+			$instance = new self( SENSEI_IB_PLUGIN_FILE, SENSEI_IB_VERSION );
+		} elseif ( class_exists( '\Sensei_Pro\Setup_Context' ) ) {
+			$instance = new self( SENSEI_PRO_PLUGIN_FILE, SENSEI_PRO_VERSION );
+		} else {
+			return;
+		}
+
+		add_filter( 'plugins_api', [ $instance, 'get_plugin_info' ], 15, 3 );
```

Comment:
As suggested in the issue, it runs after Sensei Pro (priority `10` in Sensei Pro vs `15` here). So if it already ran properly in Sensei Pro, we skip it (see inside the methods).